jobs/listen_pulse.py

```diff
@@ -1,142 +0,0 @@
-    # https://github.com/mozilla/pulsetranslator#routing-keys
-    buildinfo = body["payload"]
-    routing_key = body["_meta"]["routing_key"]
-
-    skip = (
-        buildinfo["tree"] in ("autoland", "try") or
-        buildinfo["buildtype"] in ("pgo", "debug") or
-        buildinfo["tree"] == "mozilla-inbound")
-    if skip:
-        logger.debug("Skip routing key '%s'" % routing_key)
-        return  # Do nothing.
-
-    logger.debug("Key '%s'=%s" % (routing_key, json.dumps(body)))
-    record = {
-        "build": {
-            "id": buildinfo["buildid"],
-            "date": epoch2iso(buildinfo["builddate"]),
-            "type": buildinfo["buildtype"]
-        },
-        "source": {
-            "revision": buildinfo["revision"],
-            "tree": buildinfo["tree"],
-            "product": buildinfo["product"],
-        },
-        "target": {
-            "platform": buildinfo["platform"],
-            "locale": buildinfo["locale"],
-            "version": url2version(buildinfo["buildurl"]),
-            "channel": None,
-        },
-        "download": {
-            "url": buildinfo["buildurl"],
-            "mimetype": None,
-            "size": None,
-        },
-        "systemaddons": None
-    }
-    r = client.create_record(record)
-    logger.info("Created record %s" % json.dumps(r))
-
-    # Tell the pulse queue that the message has been processed and
-    # that it is safe to discard.
-    message.ack()
-
-    # XXX: Python 3 async
-    update_download_info(client, r)
-
-
-if __name__ == "__main__":
-    parser = cli_utils.add_parser_options(
-        description="Send Pulse data to Kinto",
-        default_server=DEFAULT_SERVER,
-        default_bucket=DEFAULT_BUCKET,
-        default_collection=DEFAULT_COLLECTION)
-
-    args = parser.parse_args(sys.argv[1:])
-
-    cli_utils.setup_logger(logger, args)
-
-    logger.info("Publish at {server}/buckets/{bucket}/collections/{collection}"
-                .format(**args.__dict__))
-
-    client = cli_utils.create_client_from_args(args)
-
-    public_perms = {"read": ["system.Everyone"]}
-    client.create_bucket(permissions=public_perms, if_not_exists=True)
-    client.create_collection(if_not_exists=True)
-
-    callback = functools.partial(pulse2kinto, client=client)
-
-    c = NormalizedBuildConsumer(user=PULSEGUARDIAN_USER,
-                                password=PULSEGUARDIAN_PASSWORD,
-                                topic=PULSE_TOPIC,
-                                callback=callback)
-
-    while True:
-        try:
-            c.listen()
-        except KeyboardInterrupt:
-            raise
-        except IOError:
-            # Sometimes there is a socket timeout. Just call listen again; this
-            # is fairly common and not worth logging.
-            pass
-        except:
-            # It is possible for rabbitmq to throw other exceptions. Just log
-            # them and move on.
-            traceback.print_exc()
```

Comment:
Yes, because it is not compatible with Python3 and right now you said you don't want to rely on poling for changes.